Code/GraphMol/Fingerprints/NAMS.h

```diff
@@ -0,0 +1,213 @@
+//
+//  Copyright (C) 2023 Rocco Moretti and other RDKit contributors
+//
+//   @@ All Rights Reserved @@
+//  This file is part of the RDKit.
+//  The contents are covered by the terms of the BSD license
+//  which is included in the file license.txt, found at the root
+//  of the RDKit source tree.
+//
+// This file implements the non-contiguous atom molecular structural similarity (NAMS)
+// method of Teixeira & Falcao (http://dx.doi.org/10.1021/ci400324u)
+// for an RDKit context
+
+#include <RDGeneral/export.h>
+#ifndef __RD_NAMS_H__
+#define __RD_NAMS_H__
+
+#include <vector>
+#include <string>
+
+namespace RDKit {
+class ROMol;
+namespace NAMS {
+
+struct NAMSParameters {
+
+  NAMSParameters() = default;
+  ~NAMSParameters();
```

Comment:
You've defined a constructor but haven't implemented it.  It's also a little odd to have a struct with private members.  Perhaps this should be a class.